javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1920,27 +1933,31 @@ public T getNext() {
             final Iterator<T> that = this;
             return new AbstractIterator<T>() {
 
-                private T next = null;
+                private T next;
```

Comment:
all tests seem to be passing without these modifications also (`takeWhile`/`takeUntil`) :/